app/assets/javascripts/setupv2.js

```diff
@@ -54,6 +71,22 @@
     $(spinner_indicator).show();
     $(failure_indicator).hide();
     $(completion_indicator).hide();
+
+    hide_retry_button();
+  };
+
+  display_retry_button = function() {
+    var retry_button = $("#retry-button");
+    if (retry_button.hasClass("d-none")) {
```

Comment:
Might be nice to abstract the `if (foo.hasClass("d-none"` functionality into a function where you can pass into the object you're checking